pygfx/renderers/wgpu/meshrender.py

```diff
@@ -757,523 +506,339 @@ def helpers(self):
-    buf_pos: (pyshader.RES_BUFFER, (1, 3), Array(f32)),
-    out_pos: (pyshader.RES_OUTPUT, "Position", vec4),
-    v_dist2center: (pyshader.RES_OUTPUT, 0, vec2),
-    v_segment_length: (pyshader.RES_OUTPUT, 1, f32),
-    v_segment_width: (pyshader.RES_OUTPUT, 2, f32),
-    v_face_idx: (pyshader.RES_OUTPUT, 3, vec4),
-    v_face_coords: (pyshader.RES_OUTPUT, 4, vec3),
-):
-    # This vertex shader uses VertexId and storage buffers instead of
-    # vertex buffers. It creates 6 vertices for each face in the mesh,
-    # drawn with triangle-list. For the faces that cross the plane, we
-    # draw a (thick) line segment with round caps (we need 6 verts for that).
-    # Other faces become degenerate triangles.
-
-    # Prepare some numbers
-    screen_factor = u_stdinfo.logical_size.xy / 2.0
-    l2p = u_stdinfo.physical_size.x / u_stdinfo.logical_size.x
-    line_width = u_material.thickness  # in logical pixels
-
-    # Get the face index, and sample the vertex indices
-    segment_index = index % 6
-    face_index = (index - segment_index) // 6
-    i1 = buf_indices[face_index * 3 + 0]
-    i2 = buf_indices[face_index * 3 + 1]
-    i3 = buf_indices[face_index * 3 + 2]
-
-    # Vertex positions of this face, in local object coordinates
-    pos1 = vec3(buf_pos[i1 * 3 + 0], buf_pos[i1 * 3 + 1], buf_pos[i1 * 3 + 2])
-    pos2 = vec3(buf_pos[i2 * 3 + 0], buf_pos[i2 * 3 + 1], buf_pos[i2 * 3 + 2])
-    pos3 = vec3(buf_pos[i3 * 3 + 0], buf_pos[i3 * 3 + 1], buf_pos[i3 * 3 + 2])
-    pos1_ = u_wobject.world_transform * vec4(pos1, 1.0)
-    pos2_ = u_wobject.world_transform * vec4(pos2, 1.0)
-    pos3_ = u_wobject.world_transform * vec4(pos3, 1.0)
-    pos1 = pos1_.xyz / pos1_.w
-    pos2 = pos2_.xyz / pos2_.w
-    pos3 = pos3_.xyz / pos3_.w
-
-    # Get the plane definition
-    plane = u_material.plane.xyzw  # ax + by + cz + d
-    n = plane.xyz  # not necessarily a unit vector
-
-    # Intersect the plane with pos 1 and 2
-    p, u = pos1.xyz, pos2.xyz - pos1.xyz
-    t1 = -(plane.x * p.x + plane.y * p.y + plane.z * p.z + plane.w) / (n @ u)
-    # Intersect the plane with pos 2 and 3
-    p, u = pos2.xyz, pos3.xyz - pos2.xyz
-    t2 = -(plane.x * p.x + plane.y * p.y + plane.z * p.z + plane.w) / (n @ u)
-    # Intersect the plane with pos 3 and 1
-    p, u = pos3.xyz, pos1.xyz - pos3.xyz
-    t3 = -(plane.x * p.x + plane.y * p.y + plane.z * p.z + plane.w) / (n @ u)
-
-    # Selectors
-    b1 = i32(t1 > 0.0) * i32(t1 < 1.0) * 4
-    b2 = i32(t2 > 0.0) * i32(t2 < 1.0) * 2
-    b3 = i32(t3 > 0.0) * i32(t3 < 1.0) * 1
-    pos_index = b1 + b2 + b3
-
-    if pos_index < 3:  # or n@u == 0
-        # Just return the same vertex, resulting in degenerate triangles
-        wpos1 = vec4(pos1, 1.0)
-        the_pos = u_stdinfo.projection_transform * u_stdinfo.cam_transform * wpos1
-        the_coord = vec2(0, 0)
-        segment_length = 0.0
+    # Put it together!
+    wgsl = shader.to_string()
+    with open(r"c:/dev/py/tmp.wgsl", "wb") as f:
```

Comment:
Probably not what you want to keep around right?

Reply:
Nice catch!